packages/react/src/components/Breadcrumb/Breadcrumb.tsx

```diff
@@ -1,80 +1,288 @@
+    const breadcrumbs = breadcrumbsRef?.current;
+    if (!breadcrumbs || !window || !document) {
+      return;
+    }
+    breadcrumbsWidth.current = breadcrumbs.offsetWidth;
+  }, []);
+
+  // Collapse or uncollapse the Context Menu depending on body width
+  useLayoutEffect(() => {
+    const breadcrumbs = breadcrumbsRef?.current;
+
+    if (!breadcrumbs || !window || !document) {
+      return;
+    }
 
-  const handleScreenResize = () => {
-    enableContextMenu(window.innerWidth <= 1024); // turn on context menu when screen is resized to a smaller size
-  };
+    const observer = new ResizeObserver((entries) => {
+      const bodyWidth = entries[0].contentRect.width;
 
+      if (breadcrumbsWidth.current >= bodyWidth / 1.5) {
+        collapseMenu(true);
+      } else {
+        collapseMenu(false);
+      }
+    });
+
+    observer.observe(document.body);
+
+    return () => {
+      observer.disconnect();
+    };
+  }, [collapseMenu]);
+
+  // Position the Context Menu right below the button
+  useLayoutEffect(() => {
+    if (!buttonRef.current || !ctxMenuRef.current) {
+      return;
+    }
+    if (isMenuCollapsed && isCtxMenuOpen) {
+      const buttonRect = buttonRef.current.getBoundingClientRect();
+      const buttonCenterX = buttonRect.left + buttonRect.width / 2;
+      const contextMenuWidth = ctxMenuRef.current.offsetWidth;
+      const navStart = buttonCenterX - contextMenuWidth / 2;
+      const navTop = buttonRect.bottom + 16;
+      setCtxMenuPosition({ start: navStart, top: navTop });
+    }
+  }, [isMenuCollapsed, isCtxMenuOpen]);
+
+  // If the ContextMenu is open, close it if the user clicks outside of it
   useEffect(() => {
-    window.addEventListener("resize", handleScreenResize);
-    handleScreenResize();
+    function handleClickOutside(event: MouseEvent) {
+      if (
+        isCtxMenuOpen &&
+        ctxMenuRef.current &&
+        !ctxMenuRef.current.contains(event.target as Node) &&
+        buttonRef.current &&
+        !buttonRef.current.contains(event.target as Node)
+      ) {
+        setIsCtxMenuOpen(false);
+      }
+    }
+    document.addEventListener("mousedown", handleClickOutside);
 
     return () => {
-      window.removeEventListener("resize", handleScreenResize);
+      document.removeEventListener("mousedown", handleClickOutside);
     };
-  }, []);
+  }, [isMenuCollapsed, isCtxMenuOpen]);
 
-  const handleContextMenuToggle = () => {
-    setContextMenuToggleOpen(!toggleMenuOpen);
-  };
+  // Close the context menu if the breadcrumbs should un-collapse
+  useEffect(() => {
+    if (isCtxMenuOpen && !isMenuCollapsed) {
+      setIsCtxMenuOpen(false);
+    }
+  }, [isMenuCollapsed, isCtxMenuOpen]);
 
-  const showContext = links.length > 4 || showContextMenu ? true : false;
+  // handle tab navigation when the context menu is open
+  useEffect(() => {
+    function handleKeyDown(event: KeyboardEvent) {
+      if (!isCtxMenuOpen || !isMenuCollapsed) {
+        return;
+      }
 
```

Comment:
Let's check if refs are initialized to be more consistent like we are doing in `useLayoutEffect`